clang/lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -1491,6 +1492,12 @@ static bool IsRecordContextStructurallyEquivalent(RecordDecl *D1,
         return false;
     }
 
+    if (auto *D1Spec = dyn_cast<ClassTemplateSpecializationDecl>(DC1)) {
+      auto *D2Spec = dyn_cast<ClassTemplateSpecializationDecl>(DC2);
```

Comment:
It would be better to check if `D2Spec` is null.

Reply:
Done.

Reply:
I see now that the old code was better, there is a previous check `DC1->getDeclKind() != DC2->getDeclKind()`, this should ensure that the type is the same.

Reply:
revert